added more context to large installations of GLPI agent (#5)
glpi-project/doc-agent: source/installation/index.rst

Large Installations
^^^^^^^^^^^^^^^^^^^

Consider a scenario where the GLPI Agent application needs to be installed on large and diverse range of Windows systems. A VBScript can be useful in this scenario to perform the following tasks:

- Install it silently taking variables from the installation script previously configured.
- Check for the presence of FusionInventory and OCS Inventory agents and uninstall them if you need it.
- Configure application settings based on `commands that can be explored here <https://glpi-agent.readthedocs.io/en/latest/installation/windows-command-line.html#command-line-parameters>`.
- Log installation progress and errors for review and analysis.

A VBScript (Visual Basic Script) is provided to deploy the installer on a network:
:download:`glpi-agent-deployment.vbs <https://raw.github.com/glpi-project/glpi-agent/develop/contrib/windows/glpi-agent-deployment.vbs>`.

In this script you'll find some variables that can be changed to your environment needings (uncomment what is commented and you need. Comment what you don't need.):

> - **GLPI Agent Version** hereby named ``SetupVersion``

Reviewer: A newline is required here.
```suggestion
In this script you'll find some variables that can be changed to your environment needings (uncomment what is commented and you need. Comment what you don't need.):

- **GLPI Agent Version** hereby named ``SetupVersion``
```

- Setup the **Location** from where the script will download the MSI hereby named ``SetupLocation``

>  - If it's HTTP, HTTPS, CIFS

Reviewer: I think it's better to provide a little explanation and the default on the same entry point.
```suggestion
- Setup the **Location** from where the script will download the MSI hereby named ``SetupLocation``. It can be a HTTP or HTTPS url, a CIFS or local folder. The default is to use the github release page url.
```

Author: Yes. You're right. It's better this way.

> - Setup the **Architecture of your systems** (if you need it to be x86, x64, or if you want to let the it be installed according to the system Architecture) hereby named ``SetupArchitecture``

Reviewer: ```suggestion
- Setup the **Architecture of your systems** (if you need it to be x86, x64, or if you want to let it be installed according to the system Architecture) hereby named ``SetupArchitecture``
```

> - **Setup Options** from the :ref:`command line paramenters <windows-command-line-parameters>`

Reviewer: Here the `:ref:` link won't work unless you create a reference entry point in the targeted page. Finally you should better use my other suggestion:
```suggestion
- **Setup Options** from the `command line parameters <windows-command-line.html#command-line-parameters>`_
```

> - **Setup Options** from the `command line paramenters <https://glpi-agent.readthedocs.io/en/latest/installation/windows-command-line.html#command-line-parameters>`

Reviewer 2: use relative path for this

```suggestion
- **Setup Options** from the `command line paramenters <windows-command-line.html#command-line-parameters>`
```

In github preview, I think it doesn't work, but maybe sphinx/readthedocs generate correct links for headings reference (aka document#headings). @g-bougard or @trasher, maybe you have an example of usage in some of our documentations ?

Reviewer: On my own, in this documentation, I often uses the `:ref:` syntax, like for the `debug` here:
https://github.com/glpi-project/doc-agent/blob/d6eb2ba3a79435a47fb72841e55e9c27a4a11312/source/bug-reporting.rst?plain=1#L29

It needs to create a ref point:
https://github.com/glpi-project/doc-agent/blob/d6eb2ba3a79435a47fb72841e55e9c27a4a11312/source/configuration.rst?plain=1#L304

This has the advantage to leave rtd to resolv itself the right page: it can be moved to another place without requiring you to change all the line referencing it.

Reviewer: In this case, you need to insert the following code to the right place:
```
.. _windows-command-line-parameters:
```

And then follow this suggestion:
```suggestion
- **Setup Options** from the :ref:`command line paramenters <windows-command-line-parameters>`
```

Reviewer: Otherwise @orthagh suggestion is not working as an underscore is missing:
```suggestion
- **Setup Options** from the `command line parameters <windows-command-line.html#command-line-parameters>`_
```

 - It is recommended to keep the ``/quiet`` parameter so the user will not be bothered with wizard or command line messages or windows

>  - If you want to follow all the steps of installations, don't add the ``\quiet`` paramenter.

Reviewer: ```suggestion
 - If you want to follow all the steps of installations, don't add the ``/quiet`` parameter.
```

- **Reconfigure**
 - You will just set this up to Yes if the current installed agent has the same version you have configured on the ``SetupVersion`` above. This option, when activated, is going to reconfigure the new options for the same Agent.

>  - It's useful when you just need to change a parameter like an GLPI_URL, for example.

Reviewer: ```suggestion
 - It's useful when you just need to change a parameter like GLPI ``SERVER`` url, for example.
```

- **Repair**
 - This option will unregister and register the MSI service DLL, and reinstall it with the options selected on the script.
 - It works just when the Setup is still installed.
- **Verbose**
 - It Enables or Disables
- **Run uninstall**

>  - Here you can uncomment the agents you want the script to uninstall (FusionInventory Agent or OCSInventory Agent)

Reviewer: Just a word to say older agents are actually deprecated.
```suggestion
 - Here you can uncomment the deprecated agents you want the script to uninstall (FusionInventory Agent or OCSInventory Agent)
```

.. hint::
   Adding this VBS Script to a Computer GPO to run on startup of computers, usually works better, since there are some users with no installation rights.

>    The msi download should be accessible for every computer that needs to execute it. If you are using a local network sharing, or a regular Github URL, you must make sure the computers are able to download it from the original source.

Reviewer: Here a new line is just missing.
```suggestion
   The msi download should be accessible for every computer that needs to execute it. If you are using a local network sharing, or a regular Github URL, you must make sure the computers are able to download it from the original source.

```

Large Installations
^^^^^^^^^^^^^^^^^^^

Consider a scenario where the GLPI Agent application needs to be installed on large and diverse range of Windows systems. A VBScript can be useful in this scenario to perform the following tasks:

- Install it silently taking variables from the installation script previously configured.
- Check for the presence of FusionInventory and OCS Inventory agents and uninstall them if you need it.

> - Configure application settings based on `commands that can be explored here <https://glpi-agent.readthedocs.io/en/latest/installation/windows-command-line.html#command-line-parameters>`.

Reviewer: I think I missed to save that suggestion in my previous review. Actually the link should be the same than the one changed on l.52:
```suggestion
- Configure application settings based on `command line parameters that can be explored here <windows-command-line.html#command-line-parameters>`_.
```
